scripts/get_liquidity_params.py

```diff
@@ -1,170 +1,200 @@
-                             "max_token_to_eth_ratio": 2.0}]
-             }
-
-midas_options = { "token": "midas",
-                  "p0": 0.0001, # 1m tokens = 100 eth
-                  "options": [{"rate": 0.01 * 0.693,
-                               "min_token_to_eth_ratio": 0.5,
-                               "max_token_to_eth_ratio": 2.0}]
-             }
-
-for token_options in [midas_options]:
-    print("*********************************************")
-
-    print("token: " + str(token_options["token"]))
-    print("current price: " + str(token_options["p0"]))
-
-    for option in token_options["options"]:
-        print("****\n" + str(option))
-
-        liq = LiquidityParams(rate=option["rate"],
-                              P_0_tokens_to_eth=token_options["p0"],
-                              num_formula_precision_bits=40,
-                              max_cap_buy_eth=3.0,
-                              max_cap_sell_eth=3.0,
-                              fee_in_precents=0.25)
-
-
-        # calculate ptoential minimal and maximal ratios according to fix amounts the rm is willing to put (100 eth, 1M tokens).
-        (minimal_pmin_ratio, maximal_pmax_ratio) = liq.calc_pmin_pmax_ratio_based_on_balances(E_0_ether=100.0, T_0_tokens=1000000)
-        # actually use pmin and pmax as fixed 0.5 and 2.0
-        (pmin_ratio, pmax_ratio) = (0.5, 2.0)
-        liq.calc_params_based_on_ratios(pmin_ratio, pmax_ratio)
-
-        #E_0_ether, T_0_tokens = liq.calc_balances_based_on_pmin_pmax(
-        #    min_token_to_eth_rate=option["min_token_to_eth_ratio"] * token_options["p0"],
-        #    max_token_to_eth_rate=option["max_token_to_eth_ratio"] * token_options["p0"]
-        #)
-
-        #liq.calc_params_based_on_ratios(option["min_token_to_eth_ratio"],
-        #                                option["max_token_to_eth_ratio"])
+from pprint import pprint
+
+
+def calc_balances_from_pmin_pmax(
+    rate,
+    price,
+    min_ratio,
+    max_ratio
+):
+
+    min_rate = min_ratio * price
+    max_rate = max_ratio * price
+
+    ether = (1 / rate) * np.log(price / min_rate)
+    tokens = (1 / rate) * (1 / price - 1 / max_rate)
+    tokens_in_eth = tokens * price
+
+    print("ether: " + str(ether))
+    print("tokens: " + str(tokens))
+    print("tokens_in_eth: " + str(tokens_in_eth))
+
+    return ether, tokens
+
+
+def calc_min_max_ratio_from_balances(
+        price,
+        rate,
+        ether,
+        tokens
+):
+
+    max_rate = price / (1 - rate * price * tokens)
+    min_rate = price / np.exp(rate * ether)
+    min_ratio = min_rate / price
+    max_ratio = max_rate / price
+    print("min_min_ratio: " + str(min_ratio))
+    print("max_max_ratio: " + str(max_ratio) + "\n")
+
+    return min_ratio, max_ratio
+
+
+def calc_rate_from_eth_balance(ether, price, min_ratio):
+    min_rate = min_ratio * price
+    rate = (1 / ether) * np.log(price / min_rate)
+    print("rate: " + str(rate))
+
+
+def calc_params_from_ratios(
```

Comment:
think we might want to offer also a simpler function that will set some defaults. maybe extract `r` from initial balances etc.